### DISCUSSION TITLE
Integration with OpenAPI to discover URL endpoints?

### DISCUSSION THREAD

**Discussion by peterbenjamin-kh** [projectdiscovery/nuclei#4609] [General]
Instead of providing individual URL(s) (e.g. `-u, -target string[]`) or an input list (e.g. `-l, -list string`), is it possible to point nuclei at an openapi endpoint (e.g. `http://127.0.0.1:8080/openapi.json`) and have nuclei discover the available URL endpoints to scan?

**Comment by peterbenjamin-kh** [comment]
I just came across [this PR](https://github.com/projectdiscovery/nuclei/pull/4477).

Correct me if I am wrong, but it seems to address the question by adding support for openapi input format, among other formats (e.g. burp, postman, ...etc).

**Reply by ehsandeep** [reply]
@pbnj-dragon right, this support is getting added for fuzzing layer, where nuclei will fuzz all the endpoint parsed from openapi specs.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
